crates/encoding/src/math.rs

```diff
@@ -172,4 +196,11 @@ mod tests {
         let output: u16 = f32_to_f16(input);
         assert_eq!(0xFC00, output); // -inf
     }
+
+    #[test]
+    fn test_f16_to_f32() {
+        const EPS: f32 = 0.001;
+        let input: f32 = std::f32::consts::PI;
+        assert!((input - f16_to_f32(f32_to_f16(input))).abs() < EPS);
```

Comment:
It would be nice to have:
1) an exported bit pattern of a 16 bit float, which we check gets the right number (i.e. test that both directions aren't wrong in the same way)
2) test that values in the extreme ranges (i.e. values of the order `60_000` and `1/60_000` ± a factor of 2, if required - and 0) are appropriately roundtripped
3) special values (i.e. NaN, infinity) roundtrip as expected. This is less important, as we don't really want to see them - maybe we should `warn!` in those cases?

Reply:
Done. I added tests for max and min value roundtrip, nan/inf roundtrip, and an explicit conversion. Perhaps `test_f16_to_f32_roundtrip` should become a proptest t but I think this is good enough for now.